### ISSUE TITLE
bug(theming): Duplicate code generated by define-dark/light-theme

### ISSUE THREAD

**Issue by mrmokwa** [angular/components#23971]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Don't know if it's intended behavior or not.
Scss generated by mat.define-dark-theme or mat.define-light-theme have duplicated keys/values/styles.

Notice the duplicated values for:
- primary
- accent
- warn
- is-dark
- foreground
- background

### Reproduction

Steps to reproduce:
1.  Create new project `ng new`
1.1 Choose SCSS as stylesheet
2.  Add angular material `ng add @angular/material`
2.1 For prebuild theme, choose custom option
3. In styles.scss file, add `@debug` $name-of-app-theme
![image](https://user-images.githubusercontent.com/12090931/142047701-00c630f1-0557-4bc7-a061-0c52c08f14c6.png)

### Expected Behavior

Should exist only one definition for
- primary
- accent
- warn
- is-dark
- foreground
- background

### Actual Behavior

Below I attached the result of `@debug` . Take a look at primary color, for example and notice the value is repeated for no reason.
The only difference is the parenthesis before.

[debug.txt](https://github.com/angular/components/files/7549121/debug.txt)

### Environment

- Angular: 13.0.1
- CDK/Material: 13.0.0
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
Setting a low priority on this, because this is only in memory, it won't result in more CSS being generated.

**Comment by devversion** [comment]
This is actually an intentional duplication to ensure backwards compatibility since we changed some of the internals of the theming API in preparation for density with MDC etc.

We should mark this old pattern as deprecated in an upcoming minor and remove this in the future. Here is the related code:

https://github.com/angular/components/blob/df7c46b82a7ecdfe74aad136e0c880e5631c7b7b/src/material/core/theming/_theming.scss#L388-L411

**Comment by devversion** [comment]
Just capturing this here: https://github.com/angular/components/pull/24835. In V15/V16 we will remove this.

**Comment by devversion** [comment]
@crisbeto Did we explicitly deprecate the `.import` and legacy theming entry-point or can we just consider this being part of the PR above?

**Comment by crisbeto** [comment]
I don't think that we've announced anywhere that the `.import` files are deprecated, but we haven't been exposing any new APIs through them.

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
